src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,78 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [user, setUser] = useState<User>();
 
-      <Loader />
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsModalLoading(true);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+    if (!currentTodo) {
+      return;
+    }
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    getUser(currentTodo.userId)
+      .then(data => setUser({ ...data }))
+      .finally(() => setIsModalLoading(false));
+  }, [currentTodo]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  const handleCloseModal = () => {
+    dispatch(setCurrentTodo(null));
+  };
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+  return (
+    <div
+      className={classNames('modal', { 'is-active': currentTodo?.id !== 0 })}
```

Comment:
The condition 'currentTodo?.id !== 0' might not be sufficient to determine if the modal should be active. Consider checking if 'currentTodo' is not null or undefined instead.